Package.swift

```diff
@@ -142,9 +142,11 @@ let package = Package(
   ]
 )
 
-// If `SWIFTCI_USE_LOCAL_DEPS` is set, we are in a CI enviornment that might disallow 
+#if swift(>=5.6)
```

Comment:
I don't understand why we have this dependency. We do not need swift-docc-plugin to use docc, just the 5.5 package manifest or later. I would like to remove it.

Reply:
I thought that we needed it to build/preview the documentation using the following command. But maybe I’m missing something.
```
swift package --disable-sandbox preview-documentation --target SwiftSyntax
```

CC @kimdv